govcd/vdc.go

```diff
@@ -196,6 +197,71 @@ func (vdc *Vdc) FindVDCNetwork(network string) (OrgVDCNetwork, error) {
 	return OrgVDCNetwork{}, fmt.Errorf("can't find VDC Network: %s", network)
 }
 
+// GetVdcNetworkByHref returns an Org VDC Network reference if the network HREF matches an existing one.
+// If no valid external network is found, it returns an nil Network reference and an error
+func (vdc *Vdc) GetVdcNetworkByHref(href string) (*OrgVDCNetwork, error) {
+
+	orgNet := NewOrgVDCNetwork(vdc.client)
+
+	_, err := vdc.client.ExecuteRequest(href, http.MethodGet,
+		"", "error retrieving org vdc network: %s", nil, orgNet.OrgVDCNetwork)
+
+	// The request was successful
+	return orgNet, err
+}
+
+// GetVdcNetworkByName returns an Org VDC Network reference if the network name matches an existing one.
+// If no valid external network is found, it returns an nil Network reference and an error
+func (vdc *Vdc) GetVdcNetworkByName(name string, refresh bool) (*OrgVDCNetwork, error) {
+	if refresh {
+		err := vdc.Refresh()
+		if err != nil {
+			return nil, fmt.Errorf("error refreshing vdc: %s", err)
+		}
+	}
+	for _, an := range vdc.Vdc.AvailableNetworks {
+		for _, reference := range an.Network {
+			if reference.Name == name {
+				return vdc.GetVdcNetworkByHref(reference.HREF)
+			}
+		}
+	}
+
+	return nil, ErrorEntityNotFound
+}
+
+// GetVdcNetworkById returns an Org VDC Network reference if the network ID matches an existing one.
+// If no valid external network is found, it returns an nil Network reference and an error
+func (vdc *Vdc) GetVdcNetworkById(id string, refresh bool) (*OrgVDCNetwork, error) {
+	if refresh {
+		err := vdc.Refresh()
+		if err != nil {
+			return nil, fmt.Errorf("error refreshing vdc: %s", err)
+		}
+	}
+	for _, an := range vdc.Vdc.AvailableNetworks {
+		for _, reference := range an.Network {
+			if reference.ID == id {
+				return vdc.GetVdcNetworkByHref(reference.HREF)
+			}
+		}
+	}
+
+	return nil, ErrorEntityNotFound
+}
+
+// GetVdcNetworkByNameOrId returns a VDC Network reference if either the network name or ID matches an existing one.
+// If no valid external network is found, it returns an empty ExternalNetwork reference and an error
+func (vdc *Vdc) GetVdcNetworkByNameOrId(identifier string, refresh bool) (*OrgVDCNetwork, error) {
```

Comment:
Hi @dataclouder , really sorry for confusion - I know we discussed this, but probably it's better to not discuss these kind of things at Friday 4pm... Now that I saw the whole spectrum in PR, to me it looks that these Org network related functions are better to be called `GetOrgVdcNetworkBy...` after all. Though longer, but more intuitive, as you don't need to process in your head what kind of VDC is it referring to.

Reply:
Renamed